File: templates/template.lua
require "io"
require "os"


function quit_error(msg)
    msg = msg or ""
    io.stderr:write(string.format("Error: %s\n", msg))
    os.exit(1)
end


function main(args)
    if #args < 1 then
        quit_error("Please provide an input file.")
    end

    local file, maybeError = io.open(args[1])
    if not file then
        quit_error("Error opening file: " .. maybeError)
    end

    for line in file:lines() do
        -- do stuff here
    end

    os.exit(0)
end


main(arg)

File: templates/template.js
#!/usr/bin/env/node

const fs = require('fs/promises');


function error(msg) {
    process.stderr.write(`Error: ${msg}\n`);
    process.exit(1);
}


async function* readLines(path) {
    const file = await fs.open(path);
    for await (const line of file.readLines()) {
        yield line;
    }
}


async function main(args) {
    if (args.length < 1) {
        error('Please specify an input file.');
    }
    for await (const line of readLines(args[0])) {
        // do stuff here
    }
    process.exit(0);
}


main(process.argv.slice(2));
